src/sesame/tokens_v2.py

```diff
@@ -144,7 +145,11 @@ def parse_token(token, get_user, scope=""):
 
     """
     token = token.encode()
-    data = base64.urlsafe_b64decode(token + b"=" * (-len(token) % 4))
+    try:
+        data = base64.urlsafe_b64decode(token + b"=" * (-len(token) % 4))
+    except binascii.Error as e:
+        logger.debug("Bad token: %s", e)
+        return
```

Comment:
The documentation of `urlsafe_b64decode` doesn't say which exceptions it can raise. It feels weird to call a function from the `base64` module and catch an exception from the `binascii` module; it looks like we're depending on implementation details of the stdlib.

There's only so much that can go wrong in this line of code, so what about:

```python
    try:
        data = base64.urlsafe_b64decode(token + b"=" * (-len(token) % 4))
    except Exception:
        logger.debug("Bad token: cannot decode; was it truncated?")
        return
```

Reply:
(needs removing `import binascii` as well)

Reply:
It is documented that the `b64decode` function in the same module raises `binascii.Error` https://docs.python.org/3/library/base64.html#base64.b64decode. Since `urlsafe_b64encode` is just a thin wrapper around `b64decode` the same documentation applies. There's even a specific spot in `b64decode` that explicitly raises `binascii.Error` https://github.com/python/cpython/blob/3.9/Lib/base64.py#L86.

I understand that it looks weird, but it is documented behavior and not an implementation detail.

Reply:
Thanks for the clarification, I see!

If `urlsafe_b64decode` cannot raise any exception than `binascii.Error`, then catching `binascii.Error` or `Exception` will not make any difference in practice.

Merely, it signals a slightly different intent: "I want to catch this error" vs. "I don't want to crash". If you look at the next 20 lines of code, they're written with the "I don't want to crash" pattern.

Not going to argue this further — I'm merging the PR.